File: python-project/experiments/discrete/RunDiscreteExperiments.py
from experiments.discrete import Exp_car_evaluation, Exp_hayes_roth, Exp_spect_heart, Exp_alarm, Exp_webkb_336, \
    Exp_news_100, Exp_coil_42, Exp_solar_flare, Exp_nursery, Exp_vote, Exp_web_phishing, Exp_breast_cancer, \
    Exp_hiv_test, Exp_zoo, Exp_balance_scale


def main():
    run = 1
    n_folds = 10
    fold_log = True
    kde_bw = "normal_reference"

    exp = Exp_hiv_test.Exp_hiv_test("hiv_test")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_hayes_roth.Exp_hayes_roth("hayes_roth")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_balance_scale.Exp_balance_scale("balance_scale")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_car_evaluation.Exp_car_evaluation("car_evaluation")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_nursery.Exp_nursery("nursery")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_breast_cancer.Exp_breast_cancer("breast_cancer")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_web_phishing.Exp_web_phishing("web_phishing")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_solar_flare.Exp_solar_flare("solar_flare")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_zoo.Exp_zoo("zoo")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_vote.Exp_vote("vote")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_spect_heart.Exp_spect_heart("spect_heart")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_alarm.Exp_alarm("alarm")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_coil_42.Exp_coil_42("coil_42")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_news_100.Exp_news_100("news_100")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_webkb_336.Exp_webkb_336("webkb_336")
    exp.run(run, n_folds, fold_log, kde_bw)


if __name__ == "__main__":
    main()

File: src/main/java/experiments/missing/discrete/Run_discrete_experiments.java
package experiments.missing.discrete;

import eu.amidst.extension.learn.parameter.InitializationTypeVBEM;
import eu.amidst.extension.learn.parameter.InitializationVBEM;
import eu.amidst.extension.learn.parameter.VBEMConfig;
import eu.amidst.extension.learn.parameter.penalizer.NoPenalizer;
import eu.amidst.extension.learn.structure.hillclimber.BayesianHcConfig;
import eu.amidst.extension.util.LogUtils;
import methods.missing.MeanImputation;
import methods.missing.MissingDiscreteMethod;

import java.util.LinkedHashSet;
import java.util.Set;

public class Run_discrete_experiments {

    public static void main(String[] args) throws Exception {
        long seed = 0;
        int nRuns = 5;
        boolean storeModels = false;

        LogUtils.LogLevel logLevel = LogUtils.LogLevel.INFO;
        int maxNumberOfParents = 3;
        Set<MissingDiscreteMethod> methods = new LinkedHashSet<>();

        /* Missing VBSEM configuration */
        InitializationVBEM initializationVBEM = new InitializationVBEM(InitializationTypeVBEM.PYRAMID, 64, 16, false);
        VBEMConfig vbemConfig = new VBEMConfig(seed, 0.01, 100, initializationVBEM, new NoPenalizer());
        BayesianHcConfig bayesianHcConfig = new BayesianHcConfig(seed, 0.01, 100);

        /* Methods */
        //methods.add(new MissingVBSEM(vbemConfig, bayesianHcConfig, 100, maxNumberOfParents));
        methods.add(new MeanImputation());

        for(double missingPercentage = 0.1; missingPercentage < 0.6; missingPercentage += 0.1) {

            /* Run experiments */
            Exp_hiv_test exp_hiv_test = new Exp_hiv_test(methods);
            exp_hiv_test.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_hayes_roth exp_hayes_roth = new Exp_hayes_roth(methods);
            exp_hayes_roth.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_balance_scale exp_balance_scale = new Exp_balance_scale(methods);
            exp_balance_scale.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_car_evaluation exp_car_evaluation = new Exp_car_evaluation(methods);
            exp_car_evaluation.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_nursery exp_nursery = new Exp_nursery(methods);
            exp_nursery.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_breast_cancer exp_breast_cancer = new Exp_breast_cancer(methods);
            exp_breast_cancer.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_web_phishing exp_web_phishing = new Exp_web_phishing(methods);
            exp_web_phishing.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_solar_flare exp_solar_flare = new Exp_solar_flare(methods);
            exp_solar_flare.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_zoo exp_zoo = new Exp_zoo(methods);
            exp_zoo.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_vote exp_vote = new Exp_vote(methods);
            exp_vote.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_spect_heart exp_spect_heart = new Exp_spect_heart(methods);
            exp_spect_heart.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_alarm exp_alarm = new Exp_alarm(methods);
            exp_alarm.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_coil_42 exp_coil_42 = new Exp_coil_42(methods);
            exp_coil_42.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_news_100 exp_news_100 = new Exp_news_100(methods);
            exp_news_100.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_webkb_336 exp_webkb_336 = new Exp_webkb_336(methods);
            exp_webkb_336.runExperiment(missingPercentage, nRuns, storeModels, logLevel);
        }
    }
}
